src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredToArmBatchProcessorImpl.java

```diff
@@ -0,0 +1,316 @@
+        boolean copyRawDataToArmSuccessful = copyRawDataToArm(
+            batchItem.getSourceEod(),
+            batchItem.getArmEod(),
+            rawFilename,
+            batchItem.getPreviousStatus(),
+            () -> {
+                batchItem.setRawFilePushSuccessful(false);
+                batchItem.undoManifestFileChange();
+                updateExternalObjectDirectoryStatusToFailed(batchItem.getArmEod(), EodHelper.failedArmRawDataStatus());
+            }
+        );
+        if (copyRawDataToArmSuccessful) {
+            batchItem.setRawFilePushSuccessful(true);
+            var archiveRecord = archiveRecordService.generateArchiveRecordInfo(batchItem.getArmEod().getId(), rawFilename);
+            batchItem.setArchiveRecord(archiveRecord);
+        }
+    }
+
+    private boolean shouldAddEntryToManifestFile(BatchItem batchItem) {
+        return equalsAnyStatus(batchItem.getPreviousStatus(), EodHelper.failedArmManifestFileStatus(), EodHelper.failedArmResponseManifestFileStatus());
+    }
+
+    private void writeManifestFile(BatchItems batchItems, File archiveRecordsFile) {
+        archiveRecordFileGenerator.generateArchiveRecords(batchItems.getArchiveRecords(), archiveRecordsFile);
+    }
+
+    protected void copyMetadataToArm(File manifestFile) {
+        try {
+            BinaryData metadataFileBinary = fileOperationService.convertFileToBinaryData(manifestFile.getAbsolutePath());
+            armDataManagementApi.saveBlobDataToArm(manifestFile.getName(), metadataFileBinary);
+        } catch (BlobStorageException e) {
+            if (e.getStatusCode() == BLOB_ALREADY_EXISTS_STATUS_CODE) {
+                log.info("Metadata BLOB already exists", e);
+            } else {
+                log.error("Failed to move BLOB metadata for file {}", manifestFile.getAbsolutePath(), e);
+                throw e;
+            }
+        } catch (Exception e) {
+            log.error("Unable to move BLOB metadata for file {}", manifestFile.getAbsolutePath(), e);
+            throw e;
+        }
+    }
+
+    private void recoverByUpdatingEodToFailedArmStatus(BatchItem batchItem) {
+        if (batchItem.getArmEod() != null) {
+            batchItem.undoManifestFileChange();
+            if (!batchItem.isRawFilePushNotNeededOrSuccessfulWhenNeeded()) {
+                updateExternalObjectDirectoryStatusToFailed(batchItem.getArmEod(), EodHelper.failedArmRawDataStatus());
+            } else {
+                updateExternalObjectDirectoryStatusToFailed(batchItem.getArmEod(), EodHelper.failedArmManifestFileStatus());
+            }
+        }
+    }
+
+    /**
+     * Contains info related to the processing of a batch item.
+     */
+    @Data
+    static class BatchItem {
+
+        private ExternalObjectDirectoryEntity sourceEod;
+        private ExternalObjectDirectoryEntity armEod;
+        private String previousManifestFile;
+        private ObjectRecordStatusEntity previousStatus;
+        private Boolean rawFilePushSuccessful;
```

Comment:
Should this have a default value of false or be a primitive boolean?

Reply:
That was on purpose. I needed a way to distinguish when the push is not needed so not attempted. And when instead it was attempted and its outcome. Rather than using an Optional<Boolean> I check when that field it's `null` to know that the push was not needed